src/client/pipeline.ts

```diff
@@ -16,23 +16,50 @@ declare global {
 
 let pipeline: Pipeline | undefined
 
-export async function startPipeline(pipelineConfig: PipelineConfig) {
+export async function startPipeline(profile: PipelineProfile): Promise<Pipeline> {
   if (pipeline) {
     pipeline.stop()
   }
 
-  const { profile, speechConfig } = pipelineConfig
+  // honor any speechConfig overrides from the top-level argument
+  const speechConfig = buildSpeechConfig(profile)
   const config = {
-    ...pipelineConfig,
-    // honor any speechConfig overrides from the top-level argument
-    speechConfig: {
-      ...profile.speechConfig,
-      ...(speechConfig || {})
-    }
+    speechConfig: speechConfig,
+    ...profile
   }
 
-  pipeline = new Pipeline(config)
-  pipeline.listen()
+  return new Promise<Pipeline>((resolve, reject) => {
+    pipeline = new Pipeline(config)
+    pipeline
+      .listen()
+      .then((pipeline: Pipeline) => {
+        resolve(pipeline)
+      })
+      .catch((error: Error) => {
+        reject(error)
+      })
+  })
```

Comment:
Looks like `pipeline.listen()` already returns a promise. This whole block can change to

```ts
pipeline = new Pipeline(config)
return pipeline.listen()
```

Reply:
Duh; that's fair.